Add documentation for caching issue (#186)
avo-hq/docs.avohq.io: docs/3.0/cache.md

Our computed system do not use MemoryStore in production because it will not be shared between multiple processes (when using Puma).
:::

> ## Problem: cached rows on table do not update when record's associations suffer changes

Reviewer: Let's not say "problem" in the title.

```suggestion
## Cached rows on the table do not update when the record's associations update
```

> :::warning
> 

Reviewer 2: ```suggestion
:::warning
```

Solution 1: touch: true on association

Solution 2: override cache_hash method on resource to take associations in consideration

> :::

Reviewer: We need to show examples of how that's achieved and not in a warning.

The structure is like so:

### Solution 1

use touch true..... <please fill in the rest>

```ruby
# example parent model with association

# example child model with association and touch true
```

### Solution 2

Avo, internally, uses the `cache_hash` method to compute the hash that will be remembered by the caching driver when displaying the rows.

You can take control and override it on that particular resource to take the association into account.

```ruby{5-13}
class Avo::Resources::User < Avo::BaseResource
  def fields
    # your fields
  end

  def cache_hash
    # record.post will now be taken under consideration 
    result = [record, file_hash, record.post]

    if parent_record.present?
      result << parent_record
    end

    result
  end
end
```

> ## Problem: cached rows on table do not update when record's associations suffer changes
> :::warning
> 
> Solution 1: touch: true on association
> 
> Solution 2: override cache_hash method on resource to take associations in consideration
> :::
> 

Reviewer 2: ```suggestion
## Problem: cached rows on table do not update when record's associations suffer changes
:::warning

Solution 1: touch: true on association
Solution 2: override cache_hash method on resource to take associations in consideration
:::

```
